openmls/src/binary_tree/array_representation.rs

```diff
@@ -1,258 +1,10 @@
-    }
-
-    /// Compute the copath path from the node with the given index to the root
-    /// node and return the vector of indices of the nodes on the copath, where
-    /// the indexing corresponds to the array representation of the underlying
-    /// binary tree.
-    pub(crate) fn copath(&self, address: &T::Address) -> Result<Vec<&T>, ABinaryTreeError> {
-        let node_index = self.index(address).ok_or(ABinaryTreeError::NodeNotFound)?;
-        let copath = copath(node_index, self.size()).map_err(|_| ABinaryTreeError::OutOfBounds)?;
-        let mut copath_nodes = Vec::new();
-        for node_index in copath {
-            copath_nodes.push(
-                self.node_by_index(node_index)
-                    .ok_or(ABinaryTreeError::OutOfBounds)?,
-            )
-        }
-        Ok(copath_nodes)
-    }
-
-    /// Given a node, return the nodes index according to the array
-    /// representation as defined in the MLS spec. If the node is not in the
-    /// tree, return `None`.
-    pub(crate) fn index(&self, address: &T::Address) -> Option<NodeIndex> {
-        self.node_map.get(address).copied()
-    }
-
-    /// Compute the lowest common ancestor of the nodes with the given indices,
-    /// where the indexing corresponds to the array representation of the
-    /// underlying binary tree. Returns an `OutOfBounds` error if either of the
-    /// indices is out of the bounds of the tree.
-    pub(crate) fn lowest_common_ancestor(
-        &self,
-        address_1: &T::Address,
-        address_2: &T::Address,
-    ) -> Result<&T, ABinaryTreeError> {
-        let node_index_1 = self
-            .index(address_1)
-            .ok_or(ABinaryTreeError::NodeNotFound)?;
-        let node_index_2 = self
-            .index(address_2)
-            .ok_or(ABinaryTreeError::NodeNotFound)?;
-        let lowest_common_ancestor = lowest_common_ancestor(node_index_1, node_index_2);
-        self.node_by_index(lowest_common_ancestor)
-            .ok_or(ABinaryTreeError::OutOfBounds)
-    }
-}
-
-implement_error! {
-    pub enum ABinaryTreeError {
-        OutOfRange = "Adding nodes exceeds the maximum possible size of the tree.",
-        NotEnoughNodes = "Not enough nodes to remove.",
-        InvalidNumberOfNodes = "The given number of nodes does not allow the creation of a full, left-balanced binary tree.",
-        OutOfBounds = "The given index is outside of the tree.",
-        AddressCollision = "Found two nodes with the same address.",
-        InvalidNode = "Can't add the default node to the tree.",
-        NodeNotFound = "Can't find the node with the given address in the tree.",
-        LibraryError = "An inconsistency in the internal state of the tree was detected.",
-    }
-}
-
-#[cfg(test)]
-impl PartialEq for ABinaryTree<u32> {
-    fn eq(&self, other: &Self) -> bool {
-        self.nodes == other.nodes
-    }
-}
+#[allow(dead_code)]
+/// FIXME: There's some dead code in treemath that will be used in treesync in
```

Comment:
What do we do about this FIXME?
If it is supposed to stay in, it needs an issue number and the dead code in treeemath should be guarded to avoid warnings.

Reply:
I've removed it here and I've also removed the `allow` one module level higher. We've got lots of warnings now, but I'll deal with them the binary tree and treesync are integrated.